Development/nmos-cpp-node/node_implementation.cpp

```diff
@@ -1410,17 +1718,31 @@ namespace impl
 // into the server instance for the NMOS Node.
 nmos::experimental::node_implementation make_node_implementation(nmos::node_model& model, slog::base_gate& gate)
 {
-    return nmos::experimental::node_implementation()
+    auto node_implementation = nmos::experimental::node_implementation()
         .on_load_server_certificates(nmos::make_load_server_certificates_handler(model.settings, gate))
         .on_load_dh_param(nmos::make_load_dh_param_handler(model.settings, gate))
         .on_load_ca_certificates(nmos::make_load_ca_certificates_handler(model.settings, gate))
         .on_system_changed(make_node_implementation_system_global_handler(model, gate)) // may be omitted if not required
         .on_registration_changed(make_node_implementation_registration_handler(gate)) // may be omitted if not required
         .on_parse_transport_file(make_node_implementation_transport_file_parser()) // may be omitted if the default is sufficient
-        .on_validate_connection_resource_patch(make_node_implementation_patch_validator()) // may be omitted if not required
+        .on_validate_connection_resource_patch(make_node_implementation_patch_validator(model)) // may be omitted if not required
         .on_resolve_auto(make_node_implementation_auto_resolver(model.settings))
         .on_set_transportfile(make_node_implementation_transportfile_setter(model.node_resources, model.settings))
         .on_connection_activated(make_node_implementation_connection_activation_handler(model, gate))
         .on_validate_channelmapping_output_map(make_node_implementation_map_validator()) // may be omitted if not required
-        .on_channelmapping_activated(make_node_implementation_channelmapping_activation_handler(gate));
+        .on_channelmapping_activated(make_node_implementation_channelmapping_activation_handler(gate))
+        .on_base_edid_changed(make_node_implementation_streamcompatibility_base_edid_put_handler(gate))
+        .on_base_edid_deleted(make_node_implementation_streamcompatibility_base_edid_delete_handler(gate))
+        .on_active_constraints_changed(make_node_implementation_streamcompatibility_active_constraints_handler(model, gate))
+        .on_validate_sender_resources_against_active_constraints(make_node_implementation_streamcompatibility_sender_validator()) // may be omitted if the default is sufficient
+        .on_validate_receiver_against_transport_file(make_node_implementation_streamcompatibility_receiver_validator());
+
+    if (impl::fields::edid_support(model.settings))
```

Comment:
Rather than having to have separate code block, could you just use ternary `?:` and pass empty `std::function` in one case?

Reply:
Thank you. Shame on me! :-) Fixed.